### ISSUE TITLE
SP5 Winforms: Not possible to invert Y-axis in on a signal plot

### ISSUE THREAD

**Issue by Jofstera** [ScottPlot/ScottPlot#4085]
Hi everyone, I am trying to recreate this graph in SP5 Winforms. 

![image](https://github.com/user-attachments/assets/1e1b1c89-64b5-45b1-b08f-99a1aea78f98)

As you can see it is a vertically orientated signal plot with an inverted y-axis. The data should also be updated in real-time. 
I did some digging and found some relevant documentation 

https://scottplot.net/cookbook/5.0/SignalXY/SignalXYVerticalInvertedY/

However this code does not work on Winforms

`// invert the vertical axis` 
`myPlot.Axes.SetLimitsY(5000, 0);` 

The class `ScottPlot.Winforms.FormsPlot` does not contain have the attributes `Axes.SetLimitsY`

Something like this for Winforms would be ideal
https://github.com/ScottPlot/ScottPlot/issues/1194#issuecomment-912154734

This is also related https://github.com/ScottPlot/ScottPlot/issues/3253#issuecomment-1926170698 (basically the feature request in point 1). 

I am happy to have a look at the source code myself if someone could point me in the right direction.

Thanks!

**Comment by StendProg** [comment]
Hi @Jofstera,

A `Control` is just a wrapper for a `Plot`. You can always access the `Plot` property of any `Control`.
```c#
formsPlot1.Plot.Axes.SetLimitsY(5000, 0);
```

**Comment by KroMignon** [comment]
Hello @Jofstera 

> The class ScottPlot.Winforms.FormsPlot does not contain have the attributes Axes.SetLimitsY

This is true, but the problem is that you are mixing-up things:
* `ScottPlot.Winforms.FormsPlot` is as graphical control, the `WinForms` backend for `ScottPlot`
* `myPlot` is a `ScottPlot.Plot` instance

To setup the plot, you have to use `Plot` member of your backend ==> `ScottPlot.Winforms.FormsPlot.Plot` => https://scottplot.net/quickstart/winforms/

for example (`formsPlot1` is supposed to be the name of the WinForms control instance):
```csharp

double[] dataX = { 1, 2, 3, 4, 5 };
double[] dataY = { 1, 4, 9, 16, 25 };

formsPlot1.Plot.Add.Scatter(dataX, dataY);
formsPlot1.Plot.Axes.SetLimitsY(5000, 0);
formsPlot1.Refresh();
```

If you plan to use realtime data with rotation, perhaps it would be better to use [`DataStreamer`](https://scottplot.net/demo/5.0/#plot-streaming-data)  or [`DataLogger`](https://scottplot.net/demo/5.0/#plot-growing-data)

**Comment by Jofstera** [comment]
@StendProg @KroMignon I see where I went wrong, thank you for your help!

I did see [DataStreamer](https://scottplot.net/demo/5.0/#plot-streaming-data) and the [DataLogger](https://scottplot.net/demo/5.0/#plot-growing-data), but decided not to use them as vertical orientation is not supported.

Foolishly, I thought that as a workaround I could use a [signal plot](https://scottplot.net/cookbook/5.0/SignalXY/SignalXYVerticalInvertedY/) and simply update the `xs` and `xy` parameters dynamically in
`var signal = myPlot.Add.SignalXY(xs, ys);` for each data point.

However arrays cannot be resized dynamically. I have tried using Lists, adding new data, and then converting that into an array like so:
`            var signalPlot = signalGraph.Plot.Add.SignalXY(xAxisDataList.ToArray(), yAxisDataList.ToArray());`
But there is no way to remove the previously added lines and I'm sure my method is probably wrong too.

How difficult would it be to add vertical rotation to  [DataStreamer](https://scottplot.net/demo/5.0/#plot-streaming-data) and [DataLogger](https://scottplot.net/demo/5.0/#plot-growing-data)?

**Comment by KroMignon** [comment]
@StendProg, there is a new version of `DataStreamer` called `DataStreamer2` which uses circular buffer but which is marked as **exprimental**.

There was an effort on it to add rotation support, but it is not finished / working.

> Experimental: Added Plottables.Experimental.DataStreamer2 plot type for displaying streaming data in a circular buffer (#3946) @drolevar


WinForms do not support control rotation (like WPF for example), so without support of rotation in `ScottPlot` it self there is no working solution for WinForms AFAIK. I don't know how many efforts it takes to add this to `DataStreamer` and/or `DataLogger`.

**Comment by swharden** [comment]
resolved in #4093
